modules/RTC/TraceablePeerConnection.js

```diff
@@ -268,6 +269,57 @@ export default function TraceablePeerConnection(
      */
     this.senderVideoMaxHeight = null;
 
+    // We currently support preferring/disabling video codecs only.
+    const getCodecMimeType
+        = codec => Object.values(CodecMimeType).find(value => value === codec.toLowerCase());
+
+    // Set the codec preference that will be applied on the SDP based on the config.js settings.
+    if (this.options.preferH264 || this.options.preferredCodec) {
```

Comment:
I think picking up the `preferredCodec` value  can be simplified to:

`let preferredCodec = getCodecMimeType(this.options.preferredCodec || (this.options.preferH264 && CodecMimeType.H264));`

Next handle any corner cases:

```
if ((browser.isFirefox() && preferredCodec === CodecMimeType.VP9) {
    preferredCodec = null;
}
```

Then do similar thing to figure out the `disabledCodec` and once you have both it will come down to:

```
if (preferredCodec || disabledCodec) {
this.codecPreference = {
            enable: disabledCodec ? false : true,
            mediaType: MediaType.VIDEO,
            mimeType: disabledCodec ? disabledCodec : preferredCodec
        };
}
```

btw. kudos for defining `CodecMimeType`